client/src/modules/events/pages/eventPage.tsx

```diff
@@ -89,16 +89,20 @@ export const EventPage: NextPage = () => {
   // It's easy to create bugs by calling arrow functions before they're defined,
   // or by calling functions that rely on variables that aren't defined yet, so
   // we define everything before it's used.
+  function isAlreadyRsvp(rsvpStatus: string | undefined) {
+    const alreadyRsvp = rsvpStatus === 'yes' || rsvpStatus === 'waitlist';
+    if (alreadyRsvp) {
+      toast({ title: 'Already RSVPed', status: 'info' });
+      return true;
+    }
+    return false;
+  }
 
-  async function onRsvp(rsvpStatus: string | undefined) {
+  async function onRsvp() {
```

Comment:
your code gave me the hint, to test `onRSVP` function without it being `async`, and didn't find issue.

Do you want to remove it, and test it? or do we open new PR for that?

Reply:
@Sboonny wouldn't the `onRSVP` function need to be `async` due to the `await joinChapter` and `await rsvpToEvent` statements inside that function?

Reply:
I meant to remove the promises except for `useConfirm`, because currently it is the only value we don't know.

I will create a new PR and test it there 👍